templates/search_derived/mobile_search_clients_daily_v1/query.sql

```diff
@@ -0,0 +1,144 @@
+);
+
+WITH core_flattened_searches AS (
+  SELECT
+    *,
+    normalize_core_search_key(searches.key) AS normalized_search_key,
+    searches.value AS search_count,
+    SAFE_SUBTRACT(UNIX_DATE(DATE(submission_timestamp)), profile_date) AS profile_age_in_days
+  FROM
+    telemetry.core
+  CROSS JOIN
+    UNNEST(
+      -- Add a null search to pings that have no searches
+      IF(ARRAY_LENGTH(searches) = 0,
+         null_search(),
+         searches
+      )) AS searches
+),
+
+-- baseline has locale but not default search engine, metrics has default search engine but not locale
+fenix_client_locales AS (
+  SELECT
+    client_info.client_id,
+    udf_mode_last(ARRAY_AGG(metrics.string.glean_baseline_locale)) AS locale
+  FROM
+    org_mozilla_fenix_stable.baseline_v1
+  WHERE
+    DATE(submission_timestamp) = @submission_date
+  GROUP BY
+    client_info.client_id
+),
+
+fenix_flattened_searches AS (
+  SELECT
+    *,
+    normalize_fenix_search_key(searches.key) AS normalized_search_key,
+    searches.value AS search_count,
+    UNIX_DATE(udf_iso8601_str_to_date(client_info.first_run_date)) AS profile_creation_date,
+    SAFE.DATE_DIFF(
+      udf_iso8601_str_to_date(ping_info.end_time),
+      udf_iso8601_str_to_date(client_info.first_run_date),
+      DAY
+    ) AS profile_age_in_days
+  FROM
+    org_mozilla_fenix_stable.metrics_v1
+  LEFT JOIN
+    fenix_client_locales ON client_id = client_info.client_id
+  CROSS JOIN
+    UNNEST(
+      -- Add a null search to pings that have no searches
+      IF(ARRAY_LENGTH(metrics.labeled_counter.metrics_search_count) = 0,
+         null_search(),
+         metrics.labeled_counter.metrics_search_count
+      )) AS searches
+),
+
+combined_search_clients AS (
+  SELECT
+    DATE(submission_timestamp) AS submission_date,
+    client_id,
+    normalized_search_key[SAFE_OFFSET(0)] AS engine,
+    normalized_search_key[SAFE_OFFSET(1)] AS source,
+    SUM(IF(ARRAY_LENGTH(normalized_search_key) = 0, 0, search_count)) AS search_count,
+    udf_mode_last(ARRAY_AGG(normalized_country_code)) AS country,
+    udf_mode_last(ARRAY_AGG(locale)) AS locale,
+    normalized_app_name AS app_name,
+    udf_mode_last(ARRAY_AGG(metadata.uri.app_version)) AS app_version,
+    udf_mode_last(ARRAY_AGG(normalized_channel)) AS channel,
+    udf_mode_last(ARRAY_AGG(normalized_os)) AS os,
+    udf_mode_last(ARRAY_AGG(normalized_os_version)) AS os_version,
+    udf_mode_last(ARRAY_AGG(default_search)) AS default_search_engine,
+    CAST(NULL AS STRING) AS default_search_engine_submission_url,
+    udf_mode_last(ARRAY_AGG(distribution_id)) AS distribution_id,
+    udf_mode_last(ARRAY_AGG(profile_date)) AS profile_creation_date,
+    udf_mode_last(ARRAY_AGG(profile_age_in_days)) AS profile_age_in_days,
+    udf_mode_last(ARRAY_AGG(sample_id)) AS sample_id
+  FROM
+    core_flattened_searches
+  GROUP BY
+    client_id,
+    submission_date,
+    engine,
+    source,
+    app_name
+  UNION ALL
```

Comment:
aggregation should be done after the union instead of before it, to reduce repeated logic, unless there is a significant performance gain from doing it this way.